config.org
#+title: doom emacs config.el
#+PROPERTY: header-args :tangle yes :padline no :noweb yes
#+STARTUP: nohideblocks

* Lexical binding
:PROPERTIES:
:ID:       fe9639f6-d041-491d-ae35-21754b196591
:END:
#+begin_src emacs-lisp
;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-


;; blank
#+end_src
* Config toggles
:PROPERTIES:
:ID:       6954237f-707f-4eaa-862b-e0bd19e452a5
:END:
#+begin_src emacs-lisp
(load! "~/.config/doom/work.el")
(load! "~/.config/doom/headless.el")
#+end_src
* Fixes
:PROPERTIES:
:ID:       6796419d-bd3e-4e58-b4a2-6ff379bb97e5
:END:
#+begin_src emacs-lisp
(progn
      (defun fix+org/insert-item-below ()
              (interactive)
              (+org/insert-item-above 1)
              (org-metadown 1))

      (map!
              :after evil-org
              :map evil-org-mode-map
              :ni [C-return]   #'fix+org/insert-item-below)

      (map!
              :after org
              :map org-mode-map
              "C-RET"      #'fix+org/insert-item-below
              [C-return]   #'fix+org/insert-item-below))

(defun my-demoted-errors-wrapper (orig-func &rest args)
  "Wrapper function to call FUNC with 'with-demoted-errors'."
  (with-demoted-errors "Error in 'func': %S"
    (apply orig-func args)))

(advice-add 'org-element-at-point :around #'my-demoted-errors-wrapper)

(load! "~/.config/emacs/modules/config/default/+evil-bindings.el")
#+end_src
* Personal information
:PROPERTIES:
:ID:       a2f58232-1e91-4f02-8d19-4265b6eb02e4
:END:
#+begin_src emacs-lisp
(setq user-full-name "Julian Lopez Carballal")
(if WORK_ENV
    (setq user-mail-address "julopezc@acciona.com")
    (setq user-mail-address "julianlpc15@gmail.com"))

 ;; Calendar starts on monday
(setq calendar-week-start-day 1)

;; blank
#+end_src
* Fonts and color themes
:PROPERTIES:
:ID:       e67365d5-3d6a-429f-9d0d-90a2fa5eddc7
:END:
#+begin_src emacs-lisp
(setq doom-font (font-spec :family "FantasqueSansM Nerd Font" :size 14))
(setq doom-unicode-font (font-spec :name "Noto Color Emoji" :size 14))
(setq org-emph-face (font-spec :family "FantasqueSansM Nerd Font" :size 14))
(set-face-attribute 'default nil :family "FantasqueSansM Nerd Font" :height 120)
(set-face-attribute 'variable-pitch nil :family "FantasqueSansM Nerd Font" :height 120)
(setq doom-emoji-fallback-font-families '("Noto Color Emoji" "Noto Emoji" "Apple Color Emoji" "Segoe UI Emoji"))

(if (not HEADLESS)
(progn
        (beacon-mode 1)
        (setq beacon-color "dark orange")
        (load! "~/.config/doom/modules/LectureNotes.el")))
;; blank
(if (string= (shell-command-to-string "/usr/bin/plasma-apply-desktoptheme --list-themes 2>/dev/null | grep current | grep light || true") "")
        (setq doom-theme 'doom-moonlight)
        (setq doom-theme 'doom-solarized-light)
        )
;; blank
;; blank
#+end_src
* Doom config
:PROPERTIES:
:ID:       4a36ba05-1c32-4766-a889-8e79b9a49c13
:END:
#+begin_src emacs-lisp
;; blank
(setq  persp-save-dir (expand-file-name (concat org-directory ".sessions")))
;; blank
#+end_src
* org and evil modules
:PROPERTIES:
:ID:       a3c1450f-e71d-4b23-a4c3-fc8aa059a30d
:END:
#+begin_src emacs-lisp
(load! "~/.config/doom/modules/org.el")

(setq org-id-locations-file "~/.config/emacs/.org-id-locations")

(load! "~/.config/doom/modules/org-clock.el")

(map!
      :after org
      :leader
      :desc (documentation 'bh/punch-in) "mcpi" #'bh/punch-in
      :desc (documentation 'bh/punch-out) "mcpo" #'bh/punch-out
      :desc (documentation 'org-timer-set-timer) "mcpp" #'org-timer-set-timer
      :desc (documentation 'org-timer-start) "mcpr" #'org-timer-start
      :desc (documentation 'org-timer-pause-or-continue) "mcps" #'org-timer-pause-or-continue
      :desc (documentation 'org-timer-stop) "mcpk" #'org-timer-stop
      )

(map! :map evil-org-mode-map
      :after evil-org
      :n "g l" #'org-down-element
      :n "]]" #'org-next-visible-heading
      :n "[[" #'org-previous-visible-heading
)

;; From https://github.com/tecosaur/emacs-config/blob/master/config.org#the-utility-of-zero-width-spaces
(map! :map org-mode-map
      :nie "M-SPC M-SPC" (cmd! (insert "\u200B")))

(load! "~/.config/doom/modules/evil.el")
(load! "~/.config/doom/modules/xournalpp.el")
;; blank
#+end_src
* org-export and org-caldav
:PROPERTIES:
:ID:       02280a29-349f-4ceb-8c92-107978262a14
:END:
#+begin_src emacs-lisp
(load! "~/.config/doom/modules/org-export.el")

(load! "~/.config/doom/modules/org-caldav.el")
;; blank
#+end_src
* org-roam
:PROPERTIES:
:ID:       9ad88b67-8280-4871-8967-2dc4b6c20773
:END:
#+begin_src emacs-lisp
(load! "~/.config/doom/modules/org-roam.el")
;; blank
#+end_src
** SPC
*** bo
:PROPERTIES:
:ID:       07e92d5e-7202-47d6-b558-6a41c399052c
:END:
#+begin_src emacs-lisp
(map!
      :after org-roam
      :leader
      "bo" #'lytex/revert-buffer-reload-roam)
;; blank
#+end_src
*** r
:PROPERTIES:
:ID:       34426c0c-b43f-49e8-b302-93348be263e8
:END:
#+begin_src emacs-lisp
(map! :after org-roam
      :leader
      :prefix ("r" . "org-roam")
#+end_src
**** o
:PROPERTIES:
:ID:       5b53e429-64ae-420d-905a-048c272796f4
:END:
Open org roam sidebar
#+begin_src emacs-lisp
      :desc (documentation 'org-roam-buffer-toggle) "o" #'org-roam-buffer-toggle
#+end_src
**** u
:PROPERTIES:
:ID:       f5852e15-fb1f-4fd1-8bfb-6ecb3039cc40
:END:
#+begin_src emacs-lisp
      ;; :desc (documentation 'org-roam-unlinked-references) "u" #'org-roam-unlinked-references
#+end_src
**** c
:PROPERTIES:
:ID:       24047197-49d6-4e1c-8af6-d1269d586b81
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-roam-capture) "c" #'org-roam-capture
#+end_src
**** i
:PROPERTIES:
:ID:       87d1f000-c2f8-4015-9a0c-588771e71694
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-roam-node-insert) "i" #'org-roam-node-insert
#+end_src
**** r
:PROPERTIES:
:ID:       88367ab9-f518-45e6-beaa-8600702536b3
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-roam-node-find) "r" #'org-roam-node-find
#+end_src
**** j
:PROPERTIES:
:ID:       8ac61973-5fc4-4e65-8836-6ebcbcb3545f
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-roam-open-buffer-at-bottom) "j" #'lytex/org-roam-open-buffer-at-bottom
#+end_src
**** n
:PROPERTIES:
:ID:       fa1c9604-e0b5-45b9-ba4b-f918d2464d04
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-open-new-buffer) "n" #'lytex/org-open-new-buffer
#+end_src
**** v
:PROPERTIES:
:ID:       4bcb5bde-195d-4845-b486-08f240313f9e
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-follow-link-vsplit) "v" #'lytex/org-follow-link-vsplit
#+end_src
**** g
:PROPERTIES:
:ID:       d2f1ea15-aed2-4e85-8803-8cb9bf0cdec6
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-roam-graph) "g" #'org-roam-graph
#+end_src
**** m
:PROPERTIES:
:ID:       317bc9e5-e44f-41be-83cb-69c9c735073e
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-remark-mark) "m" #'org-remark-mark
#+end_src
**** ,
:PROPERTIES:
:ID:       0b83a266-ae0f-4699-afe5-21873593e78e
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-remark-view-prev) "," #'(lambda () (interactive) (org-remark-view-prev) (evil-scroll-line-to-center (line-number-at-pos)))
#+end_src
**** .
:PROPERTIES:
:ID:       1434d322-c18c-41ed-a518-6b12da0ca9ae
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-remark-view-next) "." #'(lambda () (interactive) (org-remark-view-next) (evil-scroll-line-to-center (line-number-at-pos)))
#+end_src
**** /
:PROPERTIES:
:ID:       36aaed9b-cfce-41bd-bb85-10120c9bddd4
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-remark-remove) "/" #'org-remark-remove
#+end_src
**** ;
:PROPERTIES:
:ID:       7aa1fb04-e93b-429d-8a01-710e1e560db9
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-remark-open) ";" #'org-remark-open)
#+end_src
*** rp
:PROPERTIES:
:ID:       a1710a4e-700b-4aab-86f5-572886418f4f
:END:
#+begin_src emacs-lisp
;; blank
(map! :after org-roam
      :leader
      :prefix ("rp" . "lytex/xournalpp")
#+end_src
**** i
:PROPERTIES:
:ID:       eaa41495-7c2d-4975-b492-488869847ae8
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/insert-global-sketch) "i" #'lytex/insert-global-sketch
#+end_src
**** p
:PROPERTIES:
:ID:       86d86ac1-c55c-4792-b427-abaae0ce0fcc
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/reset-sketch) "p" #'lytex/reset-sketch)

#+end_src
*** rf
:PROPERTIES:
:ID:       7c00f2db-d6d6-4a71-8985-1a7a89cec0d2
:END:
#+begin_src emacs-lisp
;; blank
(map! :after org-roam
      :leader
      :prefix ("rf" . "lytex/org-roam-refile")
#+end_src
**** l
:PROPERTIES:
:ID:       6e9eab60-e404-47f6-a0a0-3015ee1b76c0
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-link-and-refile) "l" #'lytex/org-link-and-refile
#+end_src
**** f
:PROPERTIES:
:ID:       05c92ddc-6bff-4634-a747-73a4f320f57d
:END:
#+begin_src emacs-lisp
      :desc (documentation #'org-roam-refile) "f" #'org-roam-refile
#+end_src
**** c
:PROPERTIES:
:ID:       53d87e65-0e7e-4f31-a2b0-35692505ab0c
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-refile-to-capture) "c" #'lytex/org-refile-to-capture
#+end_src
**** o
:PROPERTIES:
:ID:       2e2ce0da-cebc-42fb-9ab9-cbf6a56cac10
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-refile-transclude-own) "o" #'lytex/org-refile-transclude-own
#+end_src
**** b
:PROPERTIES:
:ID:       b9ab6261-ec50-46fe-b4f1-adef65df33b4
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-link-and-refile-to-capture) "b" #'lytex/org-link-and-refile-to-capture)
#+end_src
*** rh
:PROPERTIES:
:ID:       a176228c-9352-4083-9dea-95ec6fba413f
:END:
#+begin_src emacs-lisp
;; blank
(map! :after org-roam
      :leader
      :prefix ("rh" . "lytex/org-roam-heading")
;; blank
#+end_src
**** o
:PROPERTIES:
:ID:       398547ee-9f67-40f5-845a-d3b8b0b9eaf9
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-roam-heading-backlinks) "o" #'lytex/org-roam-heading-backlinks
#+end_src
**** u
:PROPERTIES:
:ID:       4eedc3e4-2431-48d8-9dd7-b859379bd9f3
:END:
#+begin_src emacs-lisp
      ;; :desc (documentation 'lytex/org-roam-heading-unlinked-references) "u" #'lytex/org-roam-heading-unlinked-references
#+end_src
**** l
:PROPERTIES:
:ID:       754ef9fa-008a-4de6-89ef-43a1a775274c
:END:
#+begin_src emacs-lisp
      :desc (documentation 'lytex/org-roam-headings-all) "l" #'lytex/org-roam-headings-all)
#+end_src
* org-transclusion
:PROPERTIES:
:ID:       0cb9603b-a760-4c8f-8ef6-5021526cd71a
:END:
#+begin_src emacs-lisp
(load! "~/.config/doom/modules/org-transclusion.el"  )

#+end_src
* org-remark
:PROPERTIES:
:ID:       5a5a6168-020f-4a3c-a252-f41407772bc6
:END:
#+begin_src emacs-lisp
(load! "~/.config/doom/modules/org-remark.el"  )
#+end_src
* org-edna
:PROPERTIES:
:ID:       8d1c0391-455e-4903-890d-e3876d794c7f
:END:
#+begin_src emacs-lisp
(if (not HEADLESS)
(progn
(load! "~/.config/doom/modules/org-edna.el")
;; blank
#+end_src
** SPC ee
:PROPERTIES:
:ID:       ae2ae410-3b98-44ee-b223-292ca9717ba4
:END:
#+begin_src emacs-lisp
;; (map!
;;       :leader
;;       :desc (documentation 'org-linker-edna) "ee" #'org-linker-edna)
;; blank
#+end_src
* org-journal
:PROPERTIES:
:ID:       24da4b74-f292-4fa1-b83f-4e3d9670d47e
:END:
#+begin_src emacs-lisp
;; blank
(load! "~/.config/doom/modules/org-journal.el")
;; blank
#+end_src
** SPC
*** om/jm
:PROPERTIES:
:ID:       1124c535-1f8f-4754-a514-a742aa43e178
:END:
#+begin_src emacs-lisp
(map!
      :after org-journal
      :leader
      "jm" #'org-journal-mode)
(map!
      :leader
      "om" #'org-mode)
;; blank
#+end_src
*** mj/mJ
:PROPERTIES:
:ID:       218c3935-6094-459d-92a7-0b8709de9a78
:END:
#+begin_src emacs-lisp
(map!
      :after org-journal
      :leader
      :desc (documentation 'org-journal-new-entry)  "mj" #'org-journal-new-entry
      :desc (documentation 'org-journal-new-scheduled-entry)  "mJ " #'org-journal-new-scheduled-entry
#+end_src
*** j
**** j
:PROPERTIES:
:ID:       fb64e2cb-1aa0-4ee9-8258-cf172abb3d6b
:END:
#+begin_src emacs-lisp
      "jj" #'lytex/set-org-journal-J
#+end_src
**** i
:PROPERTIES:
:ID:       b0cf03c8-175e-41fe-817b-97f74016b6a6
:END:
#+begin_src emacs-lisp
      "ji" #'lytex/set-org-journal-I
#+end_src
**** w
:PROPERTIES:
:ID:       a042ac59-cb22-4551-99b5-44a7d85dec90
:END:
#+begin_src emacs-lisp
      "jw" #'lytex/set-org-journal-W)
#+end_src
**** p
:PROPERTIES:
:ID:       30da5671-8254-44e9-9272-272a2d86c942
:END:
#+begin_src emacs-lisp
;; blank
(map! :after org-journal
      :leader
      :desc (documentation 'org-journal-previous-entry) "jp" #'org-journal-previous-entry
#+end_src
**** n
:PROPERTIES:
:ID:       a617e02f-5dfd-45c1-b456-f3955a577a7d
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-journal-next-entry) "jn" #'org-journal-next-entry)
#+end_src
* org-misc
:PROPERTIES:
:ID:       2726f6d9-1538-4385-9d88-2742fb3d9c45
:END:
#+begin_src emacs-lisp
;; blank
(load! "~/.config/doom/modules/org-misc.el")
;; blank
#+end_src
* org-ql
:PROPERTIES:
:ID:       a9a417e5-1971-4669-ba71-96249341c7b9
:END:
#+begin_src emacs-lisp
;; blank
;; blank
#+end_src
#+begin_src emacs-lisp
(load! "~/.config/doom/modules/org-ql.el")
;; blank
#+end_src
** SPC
*** oq
:PROPERTIES:
:ID:       12c2c90c-a7a7-4f5d-a8c7-86b5dc72f489
:END:
#+begin_src emacs-lisp
(map!
      :after org-ql
      :leader
      :prefix ("oq" . "org-ql-search")

#+end_src
**** w
:PROPERTIES:
:ID:       bf63d88d-8623-4636-ac2b-cf28984202f5
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-ql-view)  "w" #'org-ql-view
#+end_src
**** s
:PROPERTIES:
:ID:       c6d877ff-2a21-42ba-801b-ea1d8abffbe5
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-ql-view-sidebar)  "s" #'org-ql-view-sidebar
#+end_src
**** t
:PROPERTIES:
:ID:       d70c55a7-7fb8-41d6-bc0b-351b5ce21d88
:END:
#+begin_src emacs-lisp
      :desc (documentation 'org-ql-sparse-tree)  "t" #'org-ql-sparse-tree
#+end_src
**** q
:PROPERTIES:
:ID:       22232dfb-ecee-44e1-820b-74ecc94725f1
:END:
q is for query
#+begin_src emacs-lisp
      :desc (documentation 'org-ql-search)  "q" #'org-ql-search
#+end_src
**** h
:PROPERTIES:
:ID:       417db2c0-927c-4bca-b623-cfa404b87b04
:END:

config.el
;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-



(load! "~/.config/doom/work.el")
(load! "~/.config/doom/headless.el")
(progn
      (defun fix+org/insert-item-below ()
              (interactive)
              (+org/insert-item-above 1)
              (org-metadown 1))

      (map!
              :after evil-org
              :map evil-org-mode-map
              :ni [C-return]   #'fix+org/insert-item-below)

      (map!
              :after org
              :map org-mode-map
              "C-RET"      #'fix+org/insert-item-below
              [C-return]   #'fix+org/insert-item-below))

(defun my-demoted-errors-wrapper (orig-func &rest args)
  "Wrapper function to call FUNC with 'with-demoted-errors'."
  (with-demoted-errors "Error in 'func': %S"
    (apply orig-func args)))

(advice-add 'org-element-at-point :around #'my-demoted-errors-wrapper)

(load! "~/.config/emacs/modules/config/default/+evil-bindings.el")
(setq user-full-name "Julian Lopez Carballal")
(if WORK_ENV
    (setq user-mail-address "julopezc@acciona.com")
    (setq user-mail-address "julianlpc15@gmail.com"))

 ;; Calendar starts on monday
(setq calendar-week-start-day 1)


(setq doom-font (font-spec :family "FantasqueSansM Nerd Font" :size 14))
(setq doom-symbol-font (font-spec :name "Noto Color Emoji" :size 14))
(setq org-emph-face (font-spec :family "FantasqueSansM Nerd Font" :size 14))
(set-face-attribute 'default nil :family "FantasqueSansM Nerd Font" :height 120)
(set-face-attribute 'variable-pitch nil :family "FantasqueSansM Nerd Font" :height 120)
(setq doom-emoji-fallback-font-families '("Noto Color Emoji" "Noto Emoji" "Apple Color Emoji" "Segoe UI Emoji"))

(if (not HEADLESS)
(progn
        (beacon-mode 1)
        (setq beacon-color "dark orange")
        (load! "~/.config/doom/modules/LectureNotes.el")))

(with-eval-after-load 'gnus
  (set-face-attribute 'gnus-group-news-low nil :inherit 'unspecified))

(if (string= (shell-command-to-string "/usr/bin/plasma-apply-desktoptheme --list-themes 2>/dev/null | grep current | grep light || true") "")
        (setq doom-theme 'doom-moonlight)
        (setq doom-theme 'doom-solarized-light)
        )



(setq  persp-save-dir (expand-file-name (concat org-directory ".sessions")))

(load! "~/.config/doom/modules/org.el")

(setq org-id-locations-file "~/.config/emacs/.org-id-locations")

(load! "~/.config/doom/modules/org-clock.el")

(map!
      :after org
      :leader
      :desc (documentation 'bh/punch-in) "mcpi" #'bh/punch-in
      :desc (documentation 'bh/punch-out) "mcpo" #'bh/punch-out
      :desc (documentation 'org-timer-set-timer) "mcpp" #'org-timer-set-timer
      :desc (documentation 'org-timer-start) "mcpr" #'org-timer-start
      :desc (documentation 'org-timer-pause-or-continue) "mcps" #'org-timer-pause-or-continue
      :desc (documentation 'org-timer-stop) "mcpk" #'org-timer-stop
      )

(map! :map evil-org-mode-map
      :after evil-org
      :n "g l" #'org-down-element
      :n "]]" #'org-next-visible-heading
      :n "[[" #'org-previous-visible-heading
)

;; From https://github.com/tecosaur/emacs-config/blob/master/config.org#the-utility-of-zero-width-spaces
(map! :map org-mode-map
      :nie "M-SPC M-SPC" (cmd! (insert "\u200B")))

(load! "~/.config/doom/modules/evil.el")
(load! "~/.config/doom/modules/xournalpp.el")

(load! "~/.config/doom/modules/org-export.el")

(load! "~/.config/doom/modules/org-caldav.el")

(load! "~/.config/doom/modules/org-roam.el")

(map!
      :after org-roam
      :leader
      "bo" #'lytex/revert-buffer-reload-roam)

(map! :after org-roam
      :leader
      :prefix ("r" . "org-roam")
      :desc (documentation 'org-roam-buffer-toggle) "o" #'org-roam-buffer-toggle
      ;; :desc (documentation 'org-roam-unlinked-references) "u" #'org-roam-unlinked-references
      :desc (documentation 'org-roam-capture) "c" #'org-roam-capture
      :desc (documentation 'org-roam-node-insert) "i" #'org-roam-node-insert
      :desc (documentation 'org-roam-node-find) "r" #'org-roam-node-find
      :desc (documentation 'lytex/org-roam-open-buffer-at-bottom) "j" #'lytex/org-roam-open-buffer-at-bottom
      :desc (documentation 'lytex/org-open-new-buffer) "n" #'lytex/org-open-new-buffer
      :desc (documentation 'lytex/org-follow-link-vsplit) "v" #'lytex/org-follow-link-vsplit
      :desc (documentation 'org-roam-graph) "g" #'org-roam-graph
      :desc (documentation 'org-remark-mark) "m" #'org-remark-mark
      :desc (documentation 'org-remark-view-prev) "," #'(lambda () (interactive) (org-remark-view-prev) (evil-scroll-line-to-center (line-number-at-pos)))
      :desc (documentation 'org-remark-view-next) "." #'(lambda () (interactive) (org-remark-view-next) (evil-scroll-line-to-center (line-number-at-pos)))
      :desc (documentation 'org-remark-remove) "/" #'org-remark-remove
      :desc (documentation 'org-remark-open) ";" #'org-remark-open)

(map! :after org-roam
      :leader
      :prefix ("rp" . "lytex/xournalpp")
      :desc (documentation 'lytex/insert-global-sketch) "i" #'lytex/insert-global-sketch
      :desc (documentation 'lytex/reset-sketch) "p" #'lytex/reset-sketch)

(map! :after org-roam
      :leader
      :prefix ("rf" . "lytex/org-roam-refile")
      :desc (documentation 'lytex/org-link-and-refile) "l" #'lytex/org-link-and-refile
      :desc (documentation #'org-roam-refile) "f" #'org-roam-refile
      :desc (documentation 'lytex/org-refile-to-capture) "c" #'lytex/org-refile-to-capture
      :desc (documentation 'lytex/org-refile-transclude-own) "o" #'lytex/org-refile-transclude-own
      :desc (documentation 'lytex/org-link-and-refile-to-capture) "b" #'lytex/org-link-and-refile-to-capture)

(map! :after org-roam
      :leader
      :prefix ("rh" . "lytex/org-roam-heading")

      :desc (documentation 'lytex/org-roam-heading-backlinks) "o" #'lytex/org-roam-heading-backlinks
      ;; :desc (documentation 'lytex/org-roam-heading-unlinked-references) "u" #'lytex/org-roam-heading-unlinked-references
      :desc (documentation 'lytex/org-roam-headings-all) "l" #'lytex/org-roam-headings-all)
(load! "~/.config/doom/modules/org-transclusion.el"  )
(load! "~/.config/doom/modules/org-remark.el"  )
(if (not HEADLESS)
(progn
(load! "~/.config/doom/modules/org-edna.el")

;; (map!
;;       :leader
;;       :desc (documentation 'org-linker-edna) "ee" #'org-linker-edna)


(load! "~/.config/doom/modules/org-journal.el")

(map!
      :after org-journal
      :leader
      "jm" #'org-journal-mode)
(map!
      :leader
      "om" #'org-mode)

(map!
      :after org-journal
      :leader
      :desc (documentation 'org-journal-new-entry)  "mj" #'org-journal-new-entry
      :desc (documentation 'org-journal-new-scheduled-entry)  "mJ " #'org-journal-new-scheduled-entry
      "jj" #'lytex/set-org-journal-J
      "ji" #'lytex/set-org-journal-I
      "jw" #'lytex/set-org-journal-W)

(map! :after org-journal
      :leader
      :desc (documentation 'org-journal-previous-entry) "jp" #'org-journal-previous-entry
      :desc (documentation 'org-journal-next-entry) "jn" #'org-journal-next-entry)

(load! "~/.config/doom/modules/org-misc.el")



(load! "~/.config/doom/modules/org-ql.el")

(map!
      :after org-ql
      :leader
      :prefix ("oq" . "org-ql-search")
      :desc (documentation 'org-ql-view)  "w" #'org-ql-view
      :desc (documentation 'org-ql-view-sidebar)  "s" #'org-ql-view-sidebar
      :desc (documentation 'org-ql-sparse-tree)  "t" #'org-ql-sparse-tree
      :desc (documentation 'org-ql-search)  "q" #'org-ql-search
      :desc (documentation 'org-hyperscheduler-open)  "h" #'org-hyperscheduler-open
      :desc (documentation 'lytex/reload-org-ql)  "r" #'lytex/reload-org-ql)

(map! :after org-ql
      :leader
      :prefix ("rt" . "lytex/org-sparse-tree")
      :desc (documentation 'lytex/org-sparse-tree-full) "j" #'lytex/org-sparse-tree-full
      :desc (documentation 'lytex/org-sparse-tree-almost-full) "k" #'lytex/org-sparse-tree-almost-full
      :desc (documentation 'lytex/org-sparse-tree-trimmed) "l" #'lytex/org-sparse-tree-trimmed
      :desc (documentation 'lytex/org-sparse-sparse-tree) ";" #'lytex/org-sparse-sparse-tree
      :desc (documentation 'lytex/org-sparse-sparse-sparse-tree) "'" #'lytex/org-sparse-sparse-sparse-tree
      :desc (documentation 'lytex/org-jira-assigned) "e" #'lytex/org-jira-assigned
)


(use-package! helm-rg)
(use-package! helm-ag
  :custom
 (helm-ag-base-command "rg --no-heading")
 (helm-ag-success-exit-status '(0 2)))
(setq helm-rg-default-directory org-directory)
(setq helm-ag--default-directory org-directory)
(map! :after helm
      :leader
      :desc (documentation 'helm-rg) "nrg" #'helm-rg
      :desc (documentation 'helm-ag) "nrh" #'helm-ag)
(use-package! pdf-info)
(if
  (file-executable-p pdf-info-epdfinfo-program)
(progn
  (load! "~/.config/doom/modules/pdf.el")
  (after! (pdf-tools)
  (map! :leader
        :mode (pdf-view-mode)
        :prefix ("a" . "annotations in pdf")
        :desc (documentation 'pdf-annot-add-markup-annotation)
        "a" #'lytex/pdf-annot-add-markup-annotation
        :desc (documentation 'pdf-annot-add-squiggly-markup-annotation)
        "g" #'lytex/pdf-annot-add-squiggly-markup-annotation
        :desc (documentation 'pdf-annot-add-highlight-markup-annotation)
        "f" #'lytex/pdf-annot-add-highlight-markup-annotation
        :desc (documentation 'pdf-annot-add-strikeout-markup-annotation)
        "s" #'lytex/pdf-annot-add-strikeout-markup-annotation
        :desc (documentation 'pdf-annot-add-underline-markup-annotation)
        "d" #'lytex/pdf-annot-add-underline-markup-annotation
        :desc (documentation 'lytex/join-org-headline-previous)
        "p" #'lytex/join-org-headline-previous
        :desc (documentation 'lytex/join-org-headline-next)
        "n" #'lytex/join-org-headline-next)


  (map! :leader
        :mode (pdf-view-mode)
        :prefix "p"
        :desc (documentation 'pdf-history-backward)  "[" #'pdf-history-backward
        :desc (documentation 'pdf-history-forward)  "]" #'pdf-history-forward))

  (after! org-noter
        (map! :leader
        :prefix ("on" . "org-noter")
        :desc (documentation 'org-noter-sync-current-note) "s" #'org-noter-sync-current-note
        :desc (documentation 'org-noter-sync-prev-note) "p" #'org-noter-sync-prev-note
        :desc (documentation 'org-noter-sync-next-note) "n" #'org-noter-sync-next-note
        :desc (documentation 'org-noter-insert-precise-note) "i" #'org-noter-insert-precise-note
        :desc (documentation 'org-noter-kill-session) "q" #'org-noter-kill-session))))

(load! "~/.config/doom/modules/anki.el")
(after! org
  (use-package! org-anki)
  (map! :leader
        :mode (org-mode)
        :prefix ("a" . "anki")
        :desc (documentation 'org-anki-cloze-dwim) "RET" #'org-anki-cloze-dwim
        :desc (documentation 'org-anki-sync-entry) "s" #'org-anki-sync-entry
        :desc (documentation 'org-anki-sync-all) "S" #'org-anki-sync-all))
 (setq tex-fontify-script nil)
 (setq font-latex-fontify-script nil)
    (use-package! highlight-indent-guides
      :hook (prog-mode . highlight-indent-guides-mode)
      :init
      (setq highlight-indent-guides-method 'character))

    (add-hook 'ediff-load-hook
                  (lambda ()
                    (set-face-background
                      ediff-current-diff-face-A  "red")
                    (set-face-background
                      ediff-current-diff-face-B "blue")))

    ;; (use-package! activity-watch-mode)
    ;; (global-activity-watch-mode)

;; (load! "~/.config/doom/habitica.el")

;; (load! "~/.config/doom/modules/excorporate.el")

;; (if WORK_ENV
;;   (use-package! org-trello))

;; (if WORK_ENV
;;   (load! "~/.config/doom/jira.el"))

(use-package! emojify)

(add-hook 'after-init-hook #'global-emojify-mode)

(defun lytex/disable-emojify (match &rest ignored)
  (or (string= match "↔") (string= match "↖") (string= match "↗") (string= match "↘") (string= match "↙")))

(add-hook 'emojify-inhibit-functions #'lytex/disable-emojify)

(use-package! vimrc-mode)
(add-to-list 'auto-mode-alist '("\\.vim\\(rc\\)?\\'" . vimrc-mode))
(defun lytex/toggle-work ()
  "Toggle work profile"
  (interactive)
  (setq WORK_ENV (not WORK_ENV))
  (lytex/reload-org-ql))
(map!
      :after org-ql
      :leader
      :desc (documentation 'lytex/toggle-work) "tw" #'lytex/toggle-work)
(map!
      :leader
      :desc (documentation #'org-fold-hide-drawer-all) "tp" #'(lambda () (interactive) (org-fold-hide-drawer-all))
      :desc (documentation #'org-cycle-hide-drawers)  "tP" #'(lambda () (interactive) (org-cycle-hide-drawers 'all)))
)
(progn
  ;; Packages I want to have in headless mode too
(use-package! ob-plantuml
  :config
  (setq org-plantuml-jar-path "/usr/share/java/plantuml/plantuml.jar"))
  )
)
;; From https://emacs.stackexchange.com/a/33344:
(defun yf/advice-list (symbol)
  (let (result)
    (advice-mapc
     (lambda (ad props)
       (push ad result))
     symbol)
    (nreverse result)))

(defun yf/kill-advice (symbol advice)
  "Kill ADVICE from SYMBOL."
  (interactive (let* ((sym (intern (completing-read "Function: " obarray #'yf/advice-list t)))
                      (advice (let ((advices-and-their-name
                                     (mapcar (lambda (ad) (cons (prin1-to-string ad)
                                                                ad))
                                             (yf/advice-list sym))))
                                (cdr (assoc (completing-read "Remove advice: " advices-and-their-name nil t)
                                            advices-and-their-name)))))
                 (list sym advice)))
  (advice-remove symbol advice))


(defun lytex/remove-all-overlays ()
  "Remove all overlays in the current buffer."
  (interactive)
  (remove-overlays (point-min) (point-max)))
;; Disable hl-mode or all faces will be the same!
(defun what-face (pos)
  (interactive "d")
  (let ((face (or (get-char-property (point) 'read-face-name)
                  (get-char-property (point) 'face))))
    (if face (message "Face: %s" face) (message "No face at %d" pos))))
;; This needs to be called at the end, otherwise I get this error:
;; Error (org-mode-hook): Error running hook "org-rainbow-tags-mode" because: (wrong-number-of-arguments #<subr min> 0)
;; File mode specification error: (doom-hook-error org-mode-hook org-rainbow-tags-mode (wrong-number-of-arguments #<subr min> 0))
(add-hook 'org-mode-hook 'org-rainbow-tags-mode)
  (progn

        (map!
                :after evil-org
                :map evil-org-mode-map
                :n [return]   #'+org/dwim-at-point)

        )

(advice-add 'org-mode :after #'(lambda ()  (setq evil-shift-width 2) (setq tab-width 2)))
